includes/class-boldgrid-inspirations-onboarding-tasks.php

```diff
@@ -0,0 +1,258 @@
+				$active_tasks[] = array(
+					'id'            => $task['id'],
+					'task_complete' => $task['task_complete'],
+				);
+				continue;
+			}
+
+			// If the active callback is a method of this class, run it.
+			if ( is_callable( array( $this, $task['active_callback'] ) ) && true === call_user_func( array( $this, $task['active_callback'] ), $install_options ) ) {
+				$active_tasks[] = array(
+					'id'            => $task['id'],
+					'task_complete' => $task['task_complete'],
+				);
+				continue;
+			}
+		}
+
+		foreach ( $active_tasks as $task ) {
+			if ( true === $task['task_complete'] ) {
+				$complete_tasks++;
+			}
+		}
+
+		// Save the active tasks.
+		update_option( $this->option_name, $active_tasks );
+		update_option( $this->configs['onboarding_progress_option'], $complete_tasks / count( $active_tasks ) );
+	}
+
+	/**
+	 * Get Task
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param string  $task_id       The task ID.
+	 * @param boolean $task_complete Whether the task is complete.
+	 *
+	 * @return array The task config array.
+	 */
+	public function get_task( $task_id, $task_complete ) {
+		$available_tasks = $this->configs['available_onboarding_tasks'];
+
+		foreach ( $available_tasks as $task ) {
+			if ( $task['id'] === $task_id ) {
+				$task['task_complete'] = $task_complete;
+				return $task;
+			}
+		}
+
+		return array();
+	}
+
+	/**
+	 * Get Cards Data
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return array An array of cards and their tasks.
+	 */
+	public function get_cards_data() {
+		$available_cards = $this->configs['available_onboarding_cards'];
+
+		$active_cards = array();
+
+		foreach ( $available_cards as $card ) {
+			$tasks         = $this->get_tasks_for_card( $card['id'] );
+			$card['tasks'] = $tasks;
+			if ( ! empty( $tasks ) ) {
+				$active_cards[] = $card;
+			}
+		}
+
+		return $active_cards;
+	}
+
+	/**
+	 * Get Tasks for card
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param string $card_id The card ID.
+	 *
+	 * @return array The tasks for the card.
+	 */
+	public function get_tasks_for_card( $card_id ) {
+		$active_tasks = get_option( $this->option_name );
+
+		if ( ! is_array( $active_tasks ) ) {
+			return array();
+		}
+
+		$tasks_for_card = array();
+
+		foreach ( $active_tasks as $active_task ) {
+			$task = $this->get_task( $active_task['id'], $active_task['task_complete'] );
+			if ( ! empty( $task ) && $task['card_id'] === $card_id ) {
+				$tasks_for_card[] = $task;
+			}
+		}
+
+		return $tasks_for_card;
+	}
+
+	/**
+	 * Get Support Tasks Data
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return array An array of support tasks.
+	 */
+	public function get_support_tasks_data() {
+		return $this->configs['available_support_tasks'];
+	}
+
+	/**
+	 * Has Legacy Headers
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param array $install_options
+	 *
+	 * @return boolean True if the theme has legacy headers, false otherwise.
+	 */
+	public function has_legacy_headers( $install_options ) {
```

Comment:
`has_legacy_headers`, `has_custom_headers`, etc... Are these classes better suited for a utility class or another existing class? What if we need to reuse them? I would never guess that this is the class I should go to to look for these types of methods.

Reply:
This class is where the actual task objects are generated from the config files, so it makes sense that this is where their logic would live. The location of these methods is also specified in the config itself. Initially I tried adding the logic directly to the config in the form of anonymous functions in the 'active_callback' key, but it didn't like serializing anonymous functions. Alternatively, these functions could be added directly to the beginning of the onboarding.config.php file if you think that'd be better.